lib/workers/testbot.js

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright 2018 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict'
+
+const _ = require('lodash')
+const Bluebird = require('bluebird')
+const Board = require('firmata')
+const drivelist = Bluebird.promisifyAll(require('drivelist'))
+const fs = Bluebird.promisifyAll(require('fs'))
+const mountutils = Bluebird.promisifyAll(require('mountutils'))
+const imageWrite = require('etcher-image-write')
+const visuals = require('resin-cli-visuals')
+
+const WRITE_DAC_REG = 0x00
+const ENABLE_VOUT_SW = 0x03
+const DISABLE_VOUT_SW = 0x04
+const ENABLE_VREG = 0x07
+const ENABLE_FAULTRST = 0x10
+const SD_RESET_ENABLE = 0x12
+const SD_RESET_DISABLE = 0x13
+
+const LED_PIN = 13
+const HW_SERIAL5 = 5
+const SD_MUX_SEL_PIN = 28
+const USB_MUX_SEL_PIN = 29
+
+/**
+* Get dev interface of the SD card
+* @function
+* @param {String} devicePath - This is not used if linux is the host OS
+*
+* @returns {String} path to the dev interface
+*/
+
+const getDevInterface = async (devicePath) => {
+  if (process.platform === 'linux') {
+    const devIdLink = '/dev/disk/by-id/usb-Generic_Ultra_HS-SD_MMC_000008264001-0:0'
```

Comment:
Can we move this to a `DEVICE_ID_LINK` constant at the top of the file?

Reply:
Moved up